Android-Sample/app/src/main/java/com/opentok/qualitystats/sample/NetworkQualityTest.java

```diff
@@ -0,0 +1,617 @@
+                .build());
+    }
+
+
+    private long calculateVideoBitrateKbps(long ssrc, long timestamp, long currentBytesSent) {
+        long videoBitrateKbps = 0;
+
+        if (prevVideoTimestamp > 0 && ssrcToPrevBytesSent.containsKey(ssrc)) {
+            long elapsedTimeMs = timestamp - prevVideoTimestamp;
+
+            if (elapsedTimeMs == 0) {
+                return videoBitrateKbps;
+            }
+
+            long previousBytesSent = ssrcToPrevBytesSent.get(ssrc);
+            long bytesSentDifference = currentBytesSent - previousBytesSent;
+
+            videoBitrateKbps = (long) ((bytesSentDifference * BITS_PER_BYTE) / (elapsedTimeMs / (double) MILLIS_TO_SECONDS));
+        }
+
+        ssrcToPrevBytesSent.put(ssrc, currentBytesSent);
+
+        return videoBitrateKbps;
+    }
+
+
+    private final Runnable stopStatsRunnable = new Runnable() {
+        @Override
+        public void run() {
+            if (mSession != null) {
+                rtcStatsHandler.removeCallbacks(rtcStatsRunnable);
+                mPublisher.setPublishVideo(false);
+                mSession.disconnect();
+                listener.onQualityTestResults(getRecommendedSetting().getRecommendedResolution());
+            }
+        }
+    };
+
+    private final PublisherKit.PublisherRtcStatsReportListener publisherRtcStatsReportListener
+            = (publisherKit, publisherRtcStats) -> {
+        for (PublisherKit.PublisherRtcStats s : publisherRtcStats) {
+            try {
+                JSONArray rtcStatsJsonArray = new JSONArray(s.jsonArrayOfReports);
+                processRtcStats(rtcStatsJsonArray, OUTBOUND_RTP);
+            } catch (JSONException e) {
+                e.printStackTrace();
+            }
+        }
+    };
+
+    private void processRtcStats(JSONArray rtcStatsJsonArray, String direction) throws JSONException {
+        List<RtcTrackStats> videoStatsList = new ArrayList<>();
+        RtcTrackStats audioStats = null;
+        double jitter = 0.0;
+        long availableOutgoingBitrate = 0;
+        long timestamp = 0;
+        double currentRoundTripTimeMs = 0;
+
+        for (int i = 0; i < rtcStatsJsonArray.length(); i++) {
+            JSONObject rtcStatObject = rtcStatsJsonArray.getJSONObject(i);
+            String statType = rtcStatObject.getString(TYPE);
+            String kind = rtcStatObject.optString(KIND, NONE);
+
+            if (direction.equals(statType) && (VIDEO.equals(kind) || AUDIO.equals(kind))) {
+                RtcTrackStats rtcTrackStats = processOutboundRtpStats(rtcStatObject);
+                if (VIDEO.equals(kind)) {
+                    videoStatsList.add(rtcTrackStats);
+                } else {
+                    audioStats = rtcTrackStats;
+                }
+                jitter = rtcStatObject.optDouble(JITTER, -1);
+            } else if (CANDIDATE_PAIR.equals(statType)) {
+                boolean isNominated = rtcStatObject.optBoolean(IS_NOMINATED, false);
```

Comment:
Can several candidates be nominated? Why not using the `selectedCandidatePairId` from the transport stats?

https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidatePairStats/selected